toolium/config_parser.py

```diff
@@ -95,6 +98,33 @@ def _update_property_from_dict(self, section, option, new_properties):
         except KeyError:
             pass
 
+    def update_toolium_system_properties(self, new_properties):
+        """ Update config properties values or add new values if property does not exist
+        Property name must be 'TOOLIUM_[SECTION]_[OPTION]' and property value must be '[Section]_[option]=value'
+        i.e. TOOLIUM_SERVER_ENABLED='Server_enabled=true'
+
+        Section and option can not be configured in property name because they must be case sensitive and, in Windows,
+        system properties are case insensitive
+
+        :param new_properties: dict with new properties values
+        """
+        for property_name, property_value in new_properties.items():
+            name_groups = re.match('^TOOLIUM_([^_]+)_([^_]+)$', property_name)
```

Comment:
Hmm, maybe I am missing something, but I think this would not work with properties that contain underscores, like [Driver] window_width, [Server] video_enabled, etc. The second group in both this regex and the next one should allow underscores to allow for it.

Reply:
you are right, underscore must be allowed in options
thanks @pabloge !